src/titiler/mosaic/titiler/mosaic/factory.py

```diff
@@ -13,43 +13,40 @@
 from fastapi import Depends, HTTPException, Path, Query
 from geojson_pydantic.features import Feature
 from geojson_pydantic.geometries import Polygon
-from morecantile import tms
+from morecantile import tms as morecantile_tms
 from morecantile.defaults import TileMatrixSets
 from pydantic import conint
-from rio_tiler.constants import MAX_THREADS
+from rio_tiler.constants import MAX_THREADS, WGS84_CRS
 from rio_tiler.io import BaseReader, MultiBandReader, MultiBaseReader, Reader
 from rio_tiler.models import Bounds
+from rio_tiler.mosaic.methods import PixelSelectionMethod
```

Comment:
FastAPI do not like `PixelSelectionMethod` in dependency and result in 

```
    raise TypeError("asdict() should be called on dataclass instances")
TypeError: asdict() should be called on dataclass instances
```

might need to revert